lib/libc/stdlib/malloc/mrs/mrs.c

```diff
@@ -1515,6 +1515,19 @@ mrs_realloc(void *ptr, size_t size)
 	mrs_debug_printf("mrs_realloc: called ptr %p ptr size %zu new size %zu\n",
 	    ptr, old_size, size);
 
+	/*
+	 * Just return the pointer if our desired size fits.
+	 *
+	 * Only try to reclaim space by coping if we'd recover at least
+	 * half of the allocated storage.  In any other case we can't
+	 * tell the difference between shrinking and linear growth into
+	 * a very large over allocation (e.g., growing into snmalloc's
+	 * power-of-two buckets by 1K).
+	 */
+	if (ptr != NULL && cheri_gettag(ptr) && cheri_getbase(ptr) == 0 &&
+	    size <= old_size && old_size - size > (old_size >> 2))
```

Comment:
`old_size >> 2` is division by 4, presumably you want to divide by 2 here?

Reply:
That was an OBE change `>> 1` is what I meant.

Reply:
*sigh* `old_size - size > (old_size >> 2)` is backwards as that's `unused_space_if_we_reuse_buffer > half_buffer_size` we want the opposite so we reallocate only if at least half the space is wasted.